q2_assembly/megahit/megahit.py

```diff
@@ -217,9 +230,22 @@ def assemble_megahit_helper(seqs, common_args) -> ContigSequencesDirFmt:
     manifest = seqs.manifest.view(pd.DataFrame)
     result = ContigSequencesDirFmt()
 
-    for samp in list(manifest.index):
-        fwd = manifest.loc[samp, "forward"]
-        rev = manifest.loc[samp, "reverse"] if paired else None
+    if co_assemble:
+        print("Co-assembling reads from all samples")
+        fwd = ""
+        rev = ""
+        for samp in list(manifest.index):
+            fwd = fwd + "," + manifest.loc[samp, "forward"]
+            rev = (rev + "," + manifest.loc[samp, "reverse"]) if paired else None
+
+        samp = "all_samples"
+        # paths from index one to avoid the first ,
+        _process_sample(samp, fwd[1:], rev[1:], common_args, result)
```

Comment:
No need for additional variables.
```suggestion
        # paths from index one to avoid the first ,
        _process_sample("all_samples", fwd[1:], rev[1:], common_args, result)
```